### ISSUE TITLE
Nix Commands Segfault when Updating Nixpkgs Input in a Flake. 

### ISSUE THREAD

**Issue by DrMkdaddy** [NixOS/nixpkgs#305640]
**Describe the bug**

All nix related commands segfault (nix eval, nix-shell, nixos-rebuild, etc) 

**Steps To Reproduce**

Step 1: Run `nixos-rebuild build-vm --flake github:DrMkdaddy/NixOS-Config#nixos`

Step 2: Sign in with the user `noor` there should not be a password. You should be immediately signed in. 

Step 3: Run any Nix command. 

**Expected behavior**

The nix command to resolve as expected (nix-shell -p package to spawn a shell with that package.) 

**nix-env (Nix) 2.18.1**

**Additional context**

Coredump is attached. 
[core.nix.1000.c108b661b37f42dba732b9220ba41c57.11861.1713651007000000.zst.gz](https://github.com/NixOS/nixpkgs/files/15049910/core.nix.1000.c108b661b37f42dba732b9220ba41c57.11861.1713651007000000.zst.gz) (I had to gunzip it.) 

The only working revision of Nixpkgs I know of is rev `12303c652b881435065a98729eb7278313041e49`

**Comment by eclairevoyant** [comment]
Are you using fish as your shell?

**Comment by DrMkdaddy** [comment]
If it happened only with fish I'd have stopped using Fish and called it a day but it happens in Bash as well.

**Comment by eclairevoyant** [comment]
I'm unable to build your config as-is:

```
$ nixos-rebuild build-vm --flake github:DrMkdaddy/NixOS-Config#nixos
building the system configuration...
error:
       … while calling the 'seq' builtin
         at /nix/store/v4pcs3nzx54m5bmxd39win0rgl2d2hbx-source/lib/modules.nix:322:18:
          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |                  ^
          323|         _module = checked (config._module);

       … while calling the 'throw' builtin
         at /nix/store/v4pcs3nzx54m5bmxd39win0rgl2d2hbx-source/lib/modules.nix:298:18:
          297|                     ''
          298|             else throw baseMsg
             |                  ^
          299|         else null;

       error: The option `nixpkgs.packageOverrides' does not exist. Definition values:
       - In `/nix/store/32qh1x2n4n7xjypklyxldigj50za1anx-source/nixos/configuration.nix': <function>
```

note that `packageOverrides` have been deprecated for years in favour of overlays

**Comment by DrMkdaddy** [comment]
Sorry I had some edits I had made a while back.  You should be able to pull my flake now since I forgot to push them.

**Comment by DrMkdaddy** [comment]
Sorry for not having reproducing the error be smooth I had entirely forgotten to reduce all friction while writing the post.

**Comment by DrMkdaddy** [comment]
Not too sure if this influences anything but here was my response from nix-info when I ran it: 

- system: `0`
 - host os: `Linux 6.8.6-lqx2, NixOS, 24.05 (Uakari), 24.05.20240419.5c24cf2`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `0`
 - nixpkgs: `not found`
 
 in the VM I built from the flake I provided. I don't know whether or not to consider it but the `nixpkgs: not found` seems odd.

**Comment by eclairevoyant** [comment]
It's due to your custom kernel. See NixOS/nix#9725.
Fixed in nix 2.21 onwards.

**Comment by DrMkdaddy** [comment]
Thank you very much for pointing this out.
